### PR TITLE
python3Packages.wat: init at 0.1.2

### PR DESCRIPTION
**Pull request by phiadaarr**
## Description of changes

Project homepage: https://igrek51.github.io/wat/

Deep inspection of Python objects. WAT is a powerful inspection tool that allows you to delve into and examine unknown objects at runtime. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/nuclear/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  nativeCheckInputs = [
+    pytestCheckHook
+    mock
+    pydantic
+    backoff
+  ];
+  pytestFlagsArray = [
+    "tests"
+    "--ignore tests/autocomplete/test_bash_install.py"
+    "--ignore tests/sublog/test_context_logger.py"
```
Comment (phiadaarr) [inline pkgs/development/python-modules/nuclear/default.nix:41]: During developing the merge request, one test fails unexpectedly - probably due to how the sandbox handles time zones. I do not have a deep understanding of this and would be happy for any help here.

The issue is described in detail here: https://github.com/igrek51/nuclear/issues/9

To trigger the issue, you have to delete `"--ignore tests/sublog/test_context_logger.py"` in the merge request.

File: pkgs/development/python-modules/nuclear/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  src = fetchFromGitHub {
+    owner = "igrek51";
+    repo = "nuclear";
+    rev = version;
+    hash = "sha256-JuO7BKmlQE6bWKqy1QvX5U4A9YkKu/4ouTSJh9R7JGo=";
+  };
+
+  build-system = [ setuptools ];
+  propagatedBuildInputs = [
```
Comment (pbsds) [inline pkgs/development/python-modules/nuclear/default.nix:27]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/nuclear/default.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/development/python-modules/nuclear/default.nix:41]: Please rephrase these in terms of `disabledTests` and `disabledTestPaths`, with comments as to why they are disabled

Comment (pbsds) [reply pkgs/development/python-modules/nuclear/default.nix:41]: It seems the test doesn't expect a trailing `Z` in the datetime. Perhaps this is locale dependent. Try something like
```nix
preCheck = ''
  export LC_ALL=C
'';
```

Comment (phiadaarr) [reply pkgs/development/python-modules/nuclear/default.nix]: I tried the `export LC_ALL=C` trick but it did not have any effect on this test. For now the test is disabled but I will try to dig deeper on this together with the author.

**Comment by pbsds** [comment]
@ofborg build python311Packages.wat python312Packages.wat

**Pull request merged**